server/src/main/java/com/sauron/model/entities/TestEntity.java

```diff
@@ -0,0 +1,54 @@
+package com.sauron.model.entities;
+
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+import javax.persistence.Table;
+import javax.validation.constraints.NotNull;
+
+@Entity
+@Table(name = "TESTENTITIES")
+public class TestEntity extends BaseEntity{
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    @Column(name = "id", unique = true, updatable = false, nullable = false)
```

Comment:
Do you really need this explicit mapping here? I thing that JPA will know that this column should be unique and not nullable when you use @Id annotation. Specifying name also seems redundant.